iree/compiler/Codegen/SPIRV/SPIRVLowerExecutableTargetPass.cpp

```diff
@@ -94,6 +95,53 @@ void SPIRVLowerExecutableTargetPass::runOnOperation() {
     }
   }
 
+  //===--------------------------------------------------------------------===//
+  // TODO: This is HACK to unblock vectorization on tensor-semantic ops. For
+  // padding, it would mean that we need to support tiling and distributing
+  // linalg.pad_tensor ops. It's quite involved and still ongoing. So for now
+  // fallback to bufferization first (which will turn it into a `linalg.copy`
+  // op) before even deducing the CodeGen configuration.
+  if (!passPipeline) {
+    OpPassManager bufferizePipeline(
```

Comment:
I dont think we need to do this. On the CPU side the way this is handled is that you decide the workload per workgroup and pipeline to use purely based on the loop ranges of the tiled loop.

In any case this seems too convoluted to follow. Would it be possible to use a separate pipeline that bufferizes first.

Acknowledge that this is a hack to get things working. Do you have an ETA of when you can do away with this hack?

Reply:
Leftover from earlier prototyping. Moved to use a separate pipeline to handle padding/copy. This should be a better structure now.